extensions/builder.md

```diff
@@ -0,0 +1,145 @@
+# Builder
+
+A **builder** is an image that MUST contain an implementation of the lifecycle, and build-time environment, and MAY contain buildpacks. Platforms SHOULD use builders to ease the build process.
+
+## Table of Contents
+
+## Required File/Directories
```

Comment:
We should require the all specified directories, app dir, platform dir, layers dir, cache dir and are writable by the stack's User.

Reply:
Resolved in 557b34f